bioimageio/core/_resource_tests.py

```diff
@@ -361,6 +397,50 @@ def _test_expected_resource_type(
     )
 
 
+class PrecisionArgs(TypedDict):
+    """
+    Arguments, both deprecated and current, for setting the precision during validation.
+    """
+
+    absolute_tolerance: float
+    relative_tolerance: float
+    decimal: Optional[int]
+
+
+def _handle_legacy_precision_args(
+    absolute_tolerance: float, relative_tolerance: float, decimal: Optional[int]
+) -> PrecisionArgs:
+    """
+    Transform the precision arguments to conform with the current implementation.
+
+    If the deprecated `decimal` argument is used it overrides the new behaviour with
+    the old behaviour.
+    """
+    # Already conforms with current implementation
+    if decimal is None:
+        return {
+            "absolute_tolerance": absolute_tolerance,
+            "relative_tolerance": relative_tolerance,
+            "decimal": decimal,
+        }
+
+    warnings.warn(
+        "The argument `decimal` has been depricated in favour of "
+        + "`relative_tolerance` and `absolute_tolerance`, with different validation "
+        + "logic, using `numpy.testing.assert_allclose, see "
+        + "'https://numpy.org/doc/stable/reference/generated/"
+        + "numpy.testing.assert_allclose.html'. Passing a value for `decimal` will "
+        + "cause validation to revert to the old behaviour."
+    )
+    # decimal overrides new behaviour,
+    #   have to convert the params to emulate old behaviour
+    return {
+        "absolute_tolerance": 10 ** (-decimal),
+        "relative_tolerance": 0,
+        "decimal": None,
```

Comment:
I think we should not return "decimal" after handling the precision args
```suggestion
```

Reply:
The reason I did this is because a few different functions are called in a chain, so the precision arguments can be transformed in the first function called and passed onto the next function, which will also "transform" the arguments (but since they are already transformed they will be returned as they are). 

I suppose this transformation could just happen in the last function in the chain, but I wanted to make it explicit how these parameters are being handled for someone looking at a each individual function. This is also because I am not totally familiar with the code. If you think it is better, I can make the suggested change and only call `_handle_legacy_precision_args` in `load_description_and_test` or possibly `_test_model_inference`.